File: app/src/main/java/ru/nekotyan2d/mirea_switcher/utils/CookieUtils.kt
package ru.nekotyan2d.mirea_switcher.utils

import android.webkit.CookieManager

object CookieUtils {
    fun parseCookies(cookieString: String?): Map<String, String> {
        if (cookieString.isNullOrBlank()) return emptyMap()
        return cookieString.split(";")
            .map { it.trim() }
            .mapNotNull {
                val parts = it.split("=", limit = 2)
                if (parts.size == 2) parts[0] to parts[1] else null
            }
            .toMap()
    }

    fun setAuthCookie(token: String) {
        CookieManager.getInstance().apply {
            setCookie(
                "https://pulse.mirea.ru",
                ".AspNetCore.Cookies=$token; Domain=.mirea.ru; Path=/; SameSite=None; Secure"
            )
            flush()
        }
    }

    fun getAuthToken(domain: String): String? {
        val cookies = CookieManager.getInstance().getCookie(domain)
        return parseCookies(cookies)[".AspNetCore.Cookies"]
    }
}

File: lib/utils/cookie_utils.dart
import 'package:flutter_inappwebview/flutter_inappwebview.dart';

class CookieUtils {
  static final _manager = CookieManager.instance();

  static Map<String, String> parseCookies(String? cookieString) {
    if(cookieString == null || cookieString.trim().isEmpty) return {};

    return Map.fromEntries(
      cookieString.split(';').map((e) => e.trim()).map((e) {
        final parts = e.split('=');
        if(parts.length < 2) return null;
        return MapEntry(parts[0], parts.sublist(1).join('='));
      }).whereType<MapEntry<String, String>>(),
    );
  }

  static Future<void> setAuthCookie(String token) async {
    await _manager.setCookie(
      url: WebUri('https://pulse.mirea.ru'),
      name: '.AspNetCore.Cookies',
      value: token,
      domain: '.mirea.ru',
      path: '/',
      sameSite: HTTPCookieSameSitePolicy.NONE,
      isSecure: true,
    );
  }
  
  static Future<String?> getAuthToken(String domain) async {
    final cookie = await _manager.getCookie(
        url: WebUri(domain),
        name: '.AspNetCore.Cookies'
    );
    return cookie?.value;
  }
}